### ISSUE TITLE
Add 15020 named ContainerPort to sidecar template

### ISSUE THREAD

**Issue by den-is** [istio/istio#47882]
**Describe the feature request**
TLDR: Please add named port 15020 to the list of containerPorts for the sidecar, in the basic sidecar template. [Similar](https://github.com/istio/istio/blob/3639a4f44f5f8ae0057b81a9004d3209acc0f6e6/manifests/charts/istio-control/istio-discovery/files/injection-template.yaml#L200-L203) to port `http-envoy-prom` `15090`. This feature can be conditional, based on meshconfig `enablePrometheusMerge`.

\<lyrics\>
I'm using PodMonitor from Prometheus-Operator similar to what you have included in [samples/addons/extras/prometheus-operator.yaml](https://github.com/istio/istio/blob/fe4b8d8cd5d55ce9db7c87a129bdcd7c2bb6baf7/samples/addons/extras/prometheus-operator.yaml)
Unfortunately, the example in samples is not bulletproof and is not 100% production-ready (but good enough). (Probably will file another issue with potential feature suggestions).
I don't need PodMonitor to always automatically determine and replace the target `address:port` based on `prometheus.io/*` annotations.
First of all, this **hack** completely defies PodMonitor nature, but most important it is error-prone, and in some situations, you will be targeting the wrong port to scrape "Envoy" metrics.
\</lyrics\>

Excerpt from my PodMonitor:
```yaml
apiVersion: monitoring.coreos.com/v1
kind: PodMonitor
metadata:
  name: envoy-stats-monitor
  namespace: istio-system
  labels:
    monitoring: istio
spec:
  selector:
    matchExpressions:
    - {key: istio-prometheus-ignore, operator: DoesNotExist}
  namespaceSelector:
    any: true
  jobLabel: envoy-stats
  podMetricsEndpoints:
  - path: /stats/prometheus
    port: http-envoy-prom
    interval: 15s
...
```

[PodMonitor](https://prometheus-operator.dev/docs/operator/api/#monitoring.coreos.com/v1.PodMonitor) is not able to scrape target by just setting integer port (`targetPort` is deprecated)
So right now I'm limited to only described containerPort `http-envoy-prom` (15090).

Or alternatives:

**Describe alternatives you've considered**
In case if I have `enablePrometheusMerge` enabled, then create another PodMonitor with the calculated target address with hardcoded port `15020`.
I.e. something like this: `__address__ = __meta_kubernetes_pod_ip:15020`

**Affected product area (please put an X in all that apply)**

[ ] Ambient
[ ] Docs
[ ] Dual Stack
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[x] Extensions and Telemetry
[ ] Security
[ ] Test and Release
[x] User Experience
[ ] Developer Infrastructure

**Affected features (please put an X in all that apply)**

[ ] Multi Cluster
[ ] Virtual Machine
[ ] Multi Control Plane

**Additional context**
